tests/rig/src/chain.rs

```diff
@@ -183,38 +170,22 @@ impl<const RANDOMIZED_TREE: bool> Chain<RANDOMIZED_TREE> {
             transactions: transactions.into(),
         };
 
-        let oracle = ForwardRunningOracle {
-            io_implementer_init_data: Some(io_implementer_init_data(Some(state_commitment))),
-            preimage_source: self.preimage_source.clone(),
-            tree: self.state_tree.clone(),
+        let block_output: BatchOutput = forward_system::run::run_batch_with_oracle_dump_ext::<
```

Comment:
Not necessarily in this PR, but we should just use the storage oracle for simulation, right?

Reply:
I did changes mechanically, so I have no idea about that part

Reply:
Ok, will discuss later.